Lib/test/test_repl.py

```diff
@@ -195,8 +195,8 @@ def bar(x):
         expected = "(30, None, [\'def foo(x):\\n\', \'    return x + 1\\n\', \'\\n\'], \'<stdin>\')"
         self.assertIn(expected, output, expected)
 
-    def test_asyncio_repl_is_ok(self):
-        assert_python_ok("-m", "asyncio")
+    def test_asyncio_repl_no_tty_fails(self):
+        assert assert_python_failure("-m", "asyncio")
```

Comment:
How do we now know that the asyncio REPL is not working? Initially, `test_asyncio_repl_is_ok` was added because the asyncio repl is not working on Windows due to a bug.

Reply:
The fact that the new test passes means that `assert_python_ok` never tested weather the repl works.

I think t should be testet without setting `stdin` on `Popen` (inside `assert_python_ok`).

I can add that test, but it will fail because of another bug I found am intending to fix soonish (report pending, easy to reproduce by running `python -m asyncio` from the main branch).

Reply:
@Eclips4 asyncio REPL is currently broken and this wasn't caught.

Reply:
> The fact that the new test passes means that `assert_python_ok` never tested weather the repl works.
> 
> I think t should be testet without setting `stdin` on `Popen` (inside `assert_python_ok`).
> 
> I can add that test, but it will fail because of another bug I found and am intending to fix soonish (report pending, easy to reproduce by running `python -m asyncio` from the main branch).

Yes, you're right, I'm also experiencing the problem with `python -m asyncio`:
```python
./python -m asyncio
asyncio REPL 3.14.0a0 (heads/main:dc03ce797a, Jul 13 2024, 15:05:30) [GCC 13.2.0] on linux
Use "await" directly instead of "asyncio.run()".
Type "help", "copyright", "credits" or "license" for more information.
>>> import asyncio
TypeError: run_multiline_interactive_console() missing 1 required positional argument: 'namespace'
Internal error, exiting asyncio REPL...
```